src/ai/backend/manager/cli/context.py

```diff
@@ -23,36 +24,39 @@
 from ai.backend.common.exception import ConfigurationError
 from ai.backend.common.types import RedisConnectionInfo, RedisProfileTarget
 from ai.backend.logging import AbstractLogger, LocalLogger, LogLevel
+from ai.backend.manager.config.bootstrap import BootstrapConfig
 from ai.backend.manager.config.loader.legacy_etcd_loader import LegacyEtcdLoader
-from ai.backend.manager.config.local import ManagerLocalConfig
-from ai.backend.manager.config.shared import ManagerSharedConfig, RedisConfig
+from ai.backend.manager.config.unified import ManagerUnifiedConfig, RedisConfig
 
 
 class CLIContext:
-    _local_config: Optional[ManagerLocalConfig]
+    _bootstrap_config: Optional[BootstrapConfig]
     _logger: AbstractLogger
 
-    def __init__(self, config_path: Path, log_level: LogLevel) -> None:
+    def __init__(self, log_level: LogLevel, config_path: Optional[Path] = None) -> None:
         self.config_path = config_path
         self.log_level = log_level
-        self._local_config = None
+        self._bootstrap_config = None
 
     @property
-    def local_config(self) -> ManagerLocalConfig:
+    def bootstrap_config(self) -> BootstrapConfig:
         # Lazy-load the configuration only when requested.
         try:
-            if self._local_config is None:
-                self._local_config = asyncio.run(
-                    ManagerLocalConfig.load(self.config_path, self.log_level)
-                )[0]
+            if self._bootstrap_config is None:
+                if self.config_path is None:
+                    self.config_path = find_config_file("manager")
+
+                self._bootstrap_config = asyncio.run(
+                    BootstrapConfig.load_from_file(self.config_path, self.log_level)
+                )
         except ConfigurationError as e:
             print(
                 "ConfigurationError: Could not read or validate the manager local config:",
                 file=sys.stderr,
             )
             print(pformat(e.invalid_data), file=sys.stderr)
             raise click.Abort()
-        return self._local_config
+        return self._bootstrap_config
```

Comment:
Don't set heavy tasks as properties

Reply:
https://github.com/lablup/backend.ai/pull/1686
Config lazy load is intentional.